File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/value/factory/DoubleValueFactory.java
package ru.citeck.ecos.records3.record.op.atts.value.factory;

import ru.citeck.ecos.records3.record.op.atts.value.AttValue;

import java.util.Collections;
import java.util.List;

public class DoubleValueFactory implements AttValueFactory<Double> {

    @Override
    public AttValue getValue(Double value) {
        return new AttValue() {
            @Override
            public String getString() {
                return value.toString();
            }

            @Override
            public Double getDouble() {
                return value;
            }

            @Override
            public Boolean getBool() {
                return value != 0;
            }
        };
    }

    @Override
    public List<Class<? extends Double>> getValueTypes() {
        return Collections.singletonList(Double.class);
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/atts/value/factory/DoubleValueFactory.kt
package ru.citeck.ecos.records3.record.atts.value.factory

import ru.citeck.ecos.records3.record.atts.value.AttValue
import java.text.DecimalFormat
import java.text.DecimalFormatSymbols
import java.util.*

class DoubleValueFactory : AttValueFactory<Double> {

    override fun getValue(value: Double): AttValue {

        return object : AttValue {

            override fun asText(): String? {
                val format = DecimalFormat("0", DecimalFormatSymbols.getInstance(Locale.ENGLISH))
                format.maximumFractionDigits = 340
                return format.format(value)
            }

            override fun asDouble(): Double {
                return value
            }

            override fun asBoolean(): Boolean {
                return value != 0.0
            }

            override fun asRaw(): Any {
                return value
            }
        }
    }

    override fun getValueTypes() = listOf(Double::class.java, java.lang.Double::class.java)
}
